integration/aggregate_compat_test.go

```diff
@@ -272,6 +284,222 @@ func TestAggregateCompatCount(t *testing.T) {
 	testAggregateStagesCompat(t, testCases)
 }
 
+func TestAggregateCompatGroupDeterministicCollections(t *testing.T) {
+	// Scalars collection is not included because aggregation groups
+	// numbers of different types for $group, and this causes output
+	// _id to be different number type between compat and target.
+	// https://github.com/FerretDB/FerretDB/issues/2184
+	//
+	// Composites, ArrayStrings, ArrayInt32s are not included
+	// because the order in compat and target can be not deterministic.
+	// Aggregation assigns BSON array to output _id, and an array with
+	// descending sort use the greatest element for comparison causing
+	// multiple documents with the same greatest element the same order,
+	// so compat and target results in different order.
+	// https://github.com/FerretDB/FerretDB/issues/2185
+
+	providers := []shareddata.Provider{
+		// shareddata.Scalars,
+
+		shareddata.Doubles,
+		shareddata.BigDoubles,
+		shareddata.Strings,
+		shareddata.Binaries,
+		shareddata.ObjectIDs,
+		shareddata.Bools,
+		shareddata.DateTimes,
+		shareddata.Nulls,
+		shareddata.Regexes,
+		shareddata.Int32s,
+		shareddata.Timestamps,
+		shareddata.Int64s,
+		shareddata.Unsets,
+		shareddata.ObjectIDKeys,
+
+		// shareddata.Composites,
+		shareddata.PostgresEdgeCases,
+
+		shareddata.DocumentsDoubles,
+		shareddata.DocumentsStrings,
+		shareddata.DocumentsDocuments,
+
+		// shareddata.ArrayStrings,
+		shareddata.ArrayDoubles,
+		// shareddata.ArrayInt32s,
+		shareddata.ArrayRegexes,
+		shareddata.ArrayDocuments,
```

Comment:
This test works just like other compat tests except the ones that must be exception, and those are excluded. See the reason above.
When you write compat test, `PostgresEdgeCases` is included in `AllProviders()` so this test includes it too. Let's not start discussing which shareddata to include or exclude each test case, if they are in `AllProviders()`, it should tested unless stated otherwise specifically :hugs:

> Do we know if removing those that are unused will affect test time in any significant way?

`PostgresEdgeCases` is included in ` AllProviders()` so it is used. The performance is just like other compat test such as compat query test, so not sure why you are asking if adding test with less provider will affect performance?

Reply:
Right now I see that those providers are used for a single test case.

Reply:
I don't think I understand this :thinking:
`PostgresEdgeCases` is used in `AllProviders()` so it's used for all compat tests, see https://github.com/FerretDB/FerretDB/blob/main/integration/shareddata/shareddata.go#L67. Is this a problem?

Reply:
No it's not! I understood this code differently 2 days ago, thanks for explanation :+1:

Reply:
That's all good!